.github/scripts/check_skip_ci.sh

```diff
@@ -13,7 +13,8 @@ set -euo pipefail
 #
 # ... `git merge-base origin/$SKIP_CHECK_BRANCH HEAD` would return commit `D`
 # `...HEAD` specifies from the common ancestor to the latest commit on the current branch (HEAD)..
-files_to_check=$(git diff --name-only "$(git merge-base origin/$SKIP_CHECK_BRANCH HEAD~)"...HEAD)
+skip_check_branch=${SKIP_CHECK_BRANCH:?SKIP_CHECK_BRANCH is required}
+files_to_check=$(git diff --name-only "$(git merge-base origin/$skip_check_branch HEAD~)"...HEAD)
```

Comment:
This subshell ignores the `set -e` at the top of the script, so this ended up setting `files_to_check` to an empty string (it seems). Some envs won't tolerate the later array expansion of that value, but some CI shells seem to allow it, so the failure never causes issues and the script does the "unsafe default" of skipping CI.

Meta-comment: we might want to consider transitioning to [paths-ignore](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore). Even though it'd be a bit more copypasta, the upside is we can't silently skip and pass tests + security scans by accident if a future bug or misconfiguration occurs.